valohai_cli/utils/ssh.py

```diff
@@ -0,0 +1,131 @@
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+from dataclasses import dataclass
+from typing import Generator, Iterator, Optional, Tuple
+
+import click
+
+from valohai_cli.api import request
+from valohai_cli.ctx import get_project
+from valohai_cli.settings import get_settings_file_name
+from valohai_cli.utils.cli_utils import prompt_from_list
+
+
+@dataclass
+class Keypair:
+    def __init__(self, public_path: pathlib.Path, private_path: pathlib.Path) -> None:
+        self.public_path = public_path
+        self.private_path = private_path
+
+    def __repr__(self) -> str:
+        return f"Keypair(public_path={self.public_path}, private_path={self.private_path})"
```

Comment:
[Dataclasses auto-generate these for you.](https://docs.python.org/3.12/library/dataclasses.html#module-dataclasses)

```suggestion
    public_path: pathlib.Path
    private_path: pathlib.Path
```